### PR TITLE
Add winnt definitions

### PR DESCRIPTION
**Pull request by skdltmxn**

### CODE REVIEW

File: src/um/winnt.rs [hunk 1]
```diff
@@ -1592,14 +3446,377 @@ ENUM!{enum JOBOBJECTINFOCLASS {
...
+pub type PPROCESSOR_GROUP_INFO = *mut PROCESSOR_GROUP_INFO;
+STRUCT!{struct GROUP_RELATIONSHIP {
+    MaximumGroupCount: WORD,
+    ActiveGroupCount: WORD,
+    Reserved: [BYTE; 20],
+    GroupInfo: [PROCESSOR_GROUP_INFO; ANYSIZE_ARRAY],
+}}
+pub type PGROUP_RELATIONSHIP = *mut GROUP_RELATIONSHIP;
+UNION2!{union SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX_u {
+    [u8; 68] [u8; 72],
```
Comment (retep998) [inline src/um/winnt.rs]: `[u32; 17] [u64; 9]`

File: src/um/winnt.rs [hunk 2]
```diff
@@ -1592,14 +3446,377 @@ ENUM!{enum JOBOBJECTINFOCLASS {
...
+    Allocated set_Allocated[1..2],
+    AllocatedToTargetProcess set_AllocatedToTargetProcess[2..3],
+    RealTime set_RealTime[3..4],
+    ReservedFlags set_ReservedFlags[4..8],
+]);
+UNION2!{union SYSTEM_CPU_SET_INFORMATION_CpuSet_u {
+    [u8; 1],
+    AllFlags AllFlags_mut: BYTE,
+    s s_mut: SYSTEM_CPU_SET_INFORMATION_CpuSet_u_s,
+}}
```
Comment (retep998) [inline src/um/winnt.rs]: Another case where the bitfield should be applied on a field directly without an extra union.

File: src/um/winnt.rs [hunk 3]
```diff
@@ -1592,14 +3446,377 @@ ENUM!{enum JOBOBJECTINFOCLASS {
...
+    CoreIndex: BYTE,
+    LastLevelCacheIndex: BYTE,
+    NumaNodeIndex: BYTE,
+    EfficiencyClass: BYTE,
+    u: SYSTEM_CPU_SET_INFORMATION_CpuSet_u,
+    Reserved: DWORD,
+    AllocationTag: DWORD64,
+}}
+UNION2!{union SYSTEM_CPU_SET_INFORMATION_u {
+    [u32; 6],
```
Comment (retep998) [inline src/um/winnt.rs]: `[u64; 3]`

File: src/um/winnt.rs [hunk 4]
```diff
@@ -1592,14 +3446,377 @@ ENUM!{enum JOBOBJECTINFOCLASS {
...
+    BitFields: DWORD,
+}}
+BITFIELD!(XSTATE_CONFIGURATION_u_s BitFields: DWORD [
+    OptimizedSave set_OptimizedSave[0..1],
+    CompactionEnabled set_CompactionEnabled[1..2],
+]);
+UNION2!{union XSTATE_CONFIGURATION_u {
+    [u32; 1],
+    ControlFlags ControlFlags_mut: DWORD,
+    s s_mut: XSTATE_CONFIGURATION_u_s,
```
Comment (retep998) [inline src/um/winnt.rs]: Extraneous union again.

File: src/um/winnt.rs [hunk 5]
```diff
@@ -1760,52 +4037,1441 @@ pub const IO_REPARSE_TAG_DEDUP: DWORD = 0x80000013;
...
+BITFIELD!(PROCESSOR_IDLESTATE_POLICY_Flags_s BitFields: WORD [
+    AllowScaling set_AllowScaling[0..1],
+    Disabled set_Disabled[1..2],
+    Reserved set_Reserved[2..16],
+]);
+UNION2!{union PROCESSOR_IDLESTATE_POLICY_Flags {
+    [u16; 1],
+    AsWORD AsWORD_mut: WORD,
+    s s_mut: PROCESSOR_IDLESTATE_POLICY_Flags_s,
+}}
```
Comment (retep998) [inline src/um/winnt.rs]: Extraneous union.

**Comment by skdltmxn** [comment]
Fixed and merged!

File: src/um/winnt.rs (hunk 1 above)
Comment (retep998) [reply src/um/winnt.rs]: Don't forget about this!

File: src/um/winnt.rs [hunk 6]
```diff
@@ -1464,34 +2864,479 @@ STRUCT!{struct IO_COUNTERS {
...
+UNION2!{union JOBOBJECT_NOTIFICATION_LIMIT_INFORMATION_2_u2 {
+    [u32; 1],
+    RateControlTolerance RateControlTolerance_mut: JOBOBJECT_RATE_CONTROL_TOLERANCE,
+    CpuRateControlTolerance CpuRateControlTolerance_mut: JOBOBJECT_RATE_CONTROL_TOLERANCE,
+}}
+UNION2!{union JOBOBJECT_NOTIFICATION_LIMIT_INFORMATION_2_u3 {
+    [u32; 1],
+    RateControlToleranceInterval RateControlToleranceInterval_mut
+        : JOBOBJECT_RATE_CONTROL_TOLERANCE_INTERVAL,
```
[the remark this answers is not in the document]
Comment (retep998) [reply src/um/winnt.rs]: Don't forget about this either!

**Comment by retep998** [comment]
You missed two comments from the first review!

**Comment by skdltmxn** [comment]
Oops, wonder why I missed those...

**Pull request merged**